pubsub/subscriptions.rb

```diff
@@ -0,0 +1,214 @@
+# Copyright 2016 Google, Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This application demonstrates how to perform basic operations on
+# subscriptions with the Google Cloud Pub/Sub API.
+#
+# For more information, see the README.md under /pubsub and the documentation at
+# https://cloud.google.com/pubsub/docs.
+
+require "google/cloud"
+
+gcloud = Google::Cloud.new
+$pubsub_client = gcloud.pubsub
+
+# [START pubsub_list_subscriptions]
+def list_subscriptions
+  # Lists all subscriptions in the current project
+  subscriptions = $pubsub_client.subscriptions
+
+  puts "Subscriptions:"
+  subscriptions.each do |subscription|
+    puts subscription.name
+  end
+end
+# [END pubsub_list_subscriptions]
+
+# [START pubsub_list_topic_subscriptions]
+def list_topic_subscriptions topic_name:
+  # References an existing topic, e.g. "my-topic"
+  topic = $pubsub_client.topic topic_name, skip_lookup: true
```

Comment:
Do not use `skip_lookup` option

To get a topic using [`google-cloud-pubsub`](https://googlecloudplatform.github.io/google-cloud-ruby/#/docs/google-cloud/v0.20.0/google/cloud/pubsub/project?method=topic-instance):

```ruby
require "google/cloud"

gcloud = Google::Cloud.new
pubsub = gcloud.pubsub

topic = pubsub.topic "existing-topic"
```